.github/workflows/liveness-branch-sync.yml

```diff
@@ -15,6 +15,6 @@ jobs:
       - name: Checkout repo
         uses: actions/checkout@24cb9080177205b6e8c946b17badbe402adc938f # v3.4.0 https://github.com/actions/checkout/commit/24cb9080177205b6e8c946b17badbe402adc938f
       - name: Create sync pull request
-        run: gh pr create -B liveness-release -H main --title 'Sync branch main to liveness-release' --body 'Created by Github action'
+        run: gh pr edit -B liveness-release || gh pr create -B liveness-release -H main --title 'Sync branch main to liveness-release' --body 'Created by Github action'
```

Comment:
I think technically `gh pr edit -B liveness-release` could technically fail, because `gh pr edit` without any arguments (even if it has some flags) [will target](https://cli.github.com/manual/gh_pr_edit) the most recent PR that has `main` as the source branch. 

For example, I just opened #3891 that has `main` as the source branch. If you try `gh pr edit -B liveness-release` now, that command will try to edit #3891 instead of the sync PR. 

To be fair, we usually do not have any PRs that has `main` as source, but I think we should tighten this up so that we aren't surprised in the future.

Reply:
I think we could use `gh pr list` and write a bash script something like:

1. Run `gh pr list -B liveness-release -H main -s open --json title --jq "length"` to get number of PRs that meet the criteria
2. If `length == 1`, don't do anything. If `length == 0`, create the PR. And if `length >1`, which should not be possible, then we exit 1.